lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -897,59 +903,116 @@ _evaluateMinimumFee
     :: Cardano.ProtocolParameters
     -> SealedTx
     -> Maybe Coin
-_evaluateMinimumFee pp tx =
-    fromCardanoLovelace <$> minFee
+_evaluateMinimumFee pp tx = do
+    anyShelleyTx <- inAnyShelleyBasedEra (cardanoTx tx)
+    pure $ withShelleyBasedBody anyShelleyTx $ \_era body ->
+        fromCardanoLovelace $
+            Cardano.evaluateTransactionFee
+                pp
+                body
+                (estimateNumberOfWitnesses body)
+                0
+
+-- | Estimate the size of the transaction (body) when fully signed.
+_estimateSignedTxSize
+    :: Cardano.IsShelleyBasedEra era
+    => Cardano.ProtocolParameters
+    -> Cardano.TxBody era
+    -> TxSize
+_estimateSignedTxSize pparams body =
+    let
+        nWits :: Word
+        nWits = estimateNumberOfWitnesses body
+
+        -- Hack which allows us to rely on the ledger to calculate the size of
+        -- witnesses:
+        feeOfWits :: Natural
+        feeOfWits = minfee nWits - minfee 0
+
+        sizeOfWits :: TxSize
+        sizeOfWits =
+            case feeOfWits `quotRem` feePerByte of
+                (n, 0) -> TxSize n
+                (_, _) -> error $ unwords
+                    [ "evaluateTransactionSize:"
```

Comment:
Since this error is triggered within `_estimateSignedTxSize`, shouldn't we use that name in the error?

```suggestion
                    [ "estimateSignedTxSize:"
```